public/ledger.js

```diff
@@ -0,0 +1,290 @@
+          p2,
+          data,
+          resolve,
+          reject
+        }
+        this.sendQueue.push(element);
+      });
+      return promise;
+    }
+  }
+
+  /**
+   * Get ledger transport in case was already set and create one if is null
+   *
+   * @return {Promise} Promise resolved when transport is created with the transport as parameter. Promise might be rejected in case of error creating the transport
+   */
+  getTransport = () => {
+    if (this.transport) {
+      return Promise.resolve(this.transport);
+    }
+
+    const promiseMethod = new Promise((resolve, reject) => {
+      const promise = Transport.create(this.timeout, this.timeout);
+      promise.then((transport) => {
+        this.transport = transport;
+        this.methods = [
+          "getVersion",
+          "getPublicKeyData",
+          "checkAddress",
+          "sendTx",
+          "getSignatures"
+        ];
+        this.transport.decorateAppAPIMethods(this, this.methods, "HTR");
+        resolve(transport);
+      }, (e) => {
+        reject(Ledger.parseLedgerError(e));
+      });
+    });
+    return promiseMethod;
+  }
+
+  /**
+   * Check if user is in the Hathor app on ledger
+   *
+   * @return {Promise} Promise resolved if user is with Hathor app opened on ledger with version as parameter. Might be rejected if user is not with Hathor opened.
+   */
+  getVersion = async () => {
+    let result;
+    try {
+      const transport = await this.getTransport();
+      result = await this.sendToLedgerOrQueue(transport, this.commands.VERSION, 0, 0);
+    } catch (e) {
+      throw Ledger.parseLedgerError(e);
+    }
+
+    if (result.length === 8 && result.slice(0, 3).equals(Buffer.from(constants.ledgerVersionMagicStringBuffer))) {
+      this.hathorAppOpened = true;
+      return result;
+    } else {
+      throw new Error('Invalid Hathor app.');
+    }
+  }
+
+  /**
+   * Get public key data that will be used to generate the xpub
+   *
+   * @return {Promise} Promise resolved with the data as parameter. Might be rejected in case of an error or if the user denies access.
+   */
+  getPublicKeyData = async () => {
+    try {
+      const transport = await this.getTransport();
+      return await this.sendToLedgerOrQueue(transport, this.commands.PUBLIC_KEY_DATA, 0, 0);
+    } catch (e) {
+      throw Ledger.parseLedgerError(e);
+    }
+  }
+
+  /**
+   * Send address command so ledger can show specific address for user to validate
+   *
+   * @param {Buffer} index Address index to be shown on ledger
+   *
+   * @return {Promise} Promise resolved when user validates that the address is correct.
+   */
+  checkAddress = async (index) => {
+    try {
+      const transport = await this.getTransport();
+      const result = await this.sendToLedgerOrQueue(transport, this.commands.ADDRESS, 0, 0, index);
+      return result;
+    } catch (e) {
+      throw Ledger.parseLedgerError(e);
+    }
+  }
+
+  sendRequest = (transport, command, p1, p2, data) => {
```

Comment:
Missing docstring.

Reply:
Removed this method as it was not used anymore.